devo/plugin.spec.yaml

```
@@ -0,0 +1,161 @@
+tags:
+- devo
+- siem
+- logs
+- cloud
+hub_tags:
+  use_cases: [threat_detection_and_response, reporting_and_analytics, data_utility, alerting_and_notifications]
+  keywords: [logs]
+  features: []
+
+types:
+  log_entry:
+    bytesTransferred:
+      title: "Bytes Transferred"
+      type: integer
+      description: "Bytes Transferred"
+      required: false
+    clientIpAddress:
+      title: "Client IP Address"
+      type: string
+      description: "Client IP address"
+      required: false
+    cookie:
+      title: "Cookie"
+      type: string
+      description: "Cookie"
+      required: false
+    eventdate:
+      title: "Event Date"
+      type: integer
+      description: "Event date"
+      required: false
+    method:
+      title: "Method"
+      type: string
+      description: "Method"
+      required: false
+    protocol:
+      title: "Protocol"
+      type: string
+      description: "Protocol"
+      required: false
+    referralUri:
+      title: "Referral URI"
+      type: string
+      description: "Referral URI"
+      required: false
+    statusCode:
+      title: "Status Code"
+      type: integer
+      description: "Status code"
+      required: false
+    timeTaken:
+      title: "Time Taken"
+      type: integer
+      description: "Time taken"
+      required: false
+    timestamp:
+      title: "Time stamp"
+      type: string
+      description: "Time stamp"
+      required: false
+    uri:
+      title: "URI"
+      type: string
+      description: "URI"
+      required: false
+    userAgent:
+      title: "User Agent"
+      type: string
+      description: "User agent"
+      required: false
+  query_result:
+    msg:
+        title: "Message"
+        type: string
+        description: "Message"
+        required: false
+    timestamp:
+        title: "Time Stamp"
+        type: integer
+        description: "Time stamp"
+        required: false
+    cid:
+        title: "CID"
+        type: string
+        description: "CID"
+        required: false
+    status:
+        title: "Status"
+        type: integer
+        description: "Status"
+        required: false
+    object:
+        title: "Log Entries"
+        type: "[]log_entry"
+        description: "Log Entries"
+        required: false
+
+connection:
+  authentication_token:
+    title: Authentication Token
+    description: Authentication Token
+    type: credential_secret_key
+    required: true
+    example: 9de5069c5afe602b2ea0a04b66beb2c0
+  region:
+    title: Region
+    description: Region
+    type: string
+    required: true
+    example: USA
+    enum:
+      - USA
+      - EU
+      - VDC (Spain)
+
+actions:
+  query_logs:
+    title: Query Logs
+    description: Run a query against the logs
+    input:
+      query:
+        title: Query
+        description: A Query. The response be limited to 200 mb of raw data or 1000 entries whichever is hit first
```

```suggestion
        description: a LINQ query. The response be limited to 200MB of raw data or 1000 entries, whichever is hit first
```
This is a LINQ query right?

---

Unsure on Linq. It doesn't respect all the linq keywords, so I thought it was better to be vague. I think they actually interpret SQL into Linq maybe.